### PR TITLE
Init fix for issue #1332

### PR DESCRIPTION
**Pull request by georgebisbas**
This Draft PR aims to fix issue #1332 . 
Opening this to trigger discussion for further improvement.
Some reasoning behind this PR:
https://devitocodes.slack.com/archives/C7JMLMSG0/p1592481498132200

To:
* [ ] squash rebase rename

### CODE REVIEW

**Comment by mloubout** [comment]
Considering 
```

eq0 = Eq(y.symbolic_max, g[0, x], implicit_dims=(time, x)) # TIME-INVARIANT
eq1 = Inc(h1[0,0], 1, implicit_dims= (time, x, y))

op = Operator([eq0, eq1], opt='noop')
```

Works fine, I think it is just a knowb in `_extract_time_invariant` that should understand when somethign should not be moved out of the loop

**Comment by georgebisbas** [comment]
> Considering
> 
> ```
> 
> eq0 = Eq(y.symbolic_max, g[0, x], implicit_dims=(time, x)) # TIME-INVARIANT
> eq1 = Inc(h1[0,0], 1, implicit_dims= (time, x, y))
> 
> op = Operator([eq0, eq1], opt='noop')
> ```
> 
> Works fine, I think it is just a knowb in `_extract_time_invariant` that should understand when somethign should not be moved out of the loop

Just my first thought: Are we going to get flop reduction if the innermost expressin is most complex?

**Comment by georgebisbas** [comment]
> Considering
> 
> ```
> 
> eq0 = Eq(y.symbolic_max, g[0, x], implicit_dims=(time, x)) # TIME-INVARIANT
> eq1 = Inc(h1[0,0], 1, implicit_dims= (time, x, y))
> 
> op = Operator([eq0, eq1], opt='noop')
> ```
> 
> Works fine, I think it is just a knowb in `_extract_time_invariant` that should understand when somethign should not be moved out of the loop
Nitpicking
For example:
If
> eq1 = Inc(h1[0,0], u*u, implicit_dims= (time, x, y))
we get pow instead if mul.

File: devito/passes/clusters/misc.py [hunk 1]
```diff
@@ -59,6 +60,34 @@ def callback(self, clusters, prefix):
...
+            # Store the symbolic bounds of dimensions in impacted clusters
+            imp_bounds = set()
+            # Store the free symbols of expressions in impacted clusters
+            imp_free_symbols = set()
+            for imp in impacted:
+                imp_free_symbols.update(imp.free_symbols)
+                for i in imp.ispace.dimensions:
+                    imp_bounds.update([i.symbolic_min, i.symbolic_max])
+
+            # Are we writing to the loop bounds of an impacted cluster?
```
Comment (FabioLuporini) [inline devito/passes/clusters/misc.py:73]: there exist no loops in this layer

**Comment by FabioLuporini** [comment]
closing as superseded by #1389

**Pull request closed without merging**